bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/S3BucketWatcherHandler.java

```diff
@@ -0,0 +1,164 @@
+            }
+        }
+        if (region == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Please enter valid region");
+            return;
+        }
+        String accessKey = config.awskey;
+        String secretKey = config.awssecret;
+        AwsCredentials credentials;
+
+        if (config.Anonymous) {
+            credentialsProvider = AnonymousCredentialsProvider.create();
+        } else {
+            if (accessKey != null && !accessKey.isBlank() && secretKey != null && !secretKey.isBlank()) {
+                credentials = AwsBasicCredentials.create(accessKey, secretKey);
+                credentialsProvider = StaticCredentialsProvider.create(credentials);
+            }
+        }
+
+        if (refreshS3BucketInformation()) {
+            if (config.pollIntervalS3 > 0) {
+                updateStatus(ThingStatus.ONLINE);
+                executionJob = scheduler.scheduleWithFixedDelay(this::refreshS3BucketInformation, config.pollIntervalS3,
+                        config.pollIntervalS3, TimeUnit.SECONDS);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Polling interval can't be null or negative");
+                return;
+            }
+        }
+    }
+
+    private boolean refreshS3BucketInformation() {
+        List<String> currentS3Listing = new ArrayList<>();
+        try {
+            client = S3Client.builder().region(region).credentialsProvider(credentialsProvider).build();
+            ListObjectsRequest listObjects = ListObjectsRequest.builder().bucket(config.S3BucketName).build();
+            ListObjectsResponse res = client.listObjects(listObjects);
+            List<S3Object> objects = res.contents();
+
+            for (S3Object s3Obj : objects) {
+                if (s3Obj.key().startsWith(config.s3path)) {
+                    currentS3Listing.add(s3Obj.key());
+                }
+            }
+            client.close();
+
+            List<String> difS3Listing = new ArrayList<>(currentS3Listing);
+            difS3Listing.removeAll(previousS3Listing);
+            difS3Listing.forEach(file -> triggerChannel(CHANNEL_NEWS3FILE, file));
+
+            if (!difS3Listing.isEmpty()) {
+                WatcherCommon.saveNewListing(difS3Listing, currentS3ListingFile);
+            }
+            previousS3Listing = new ArrayList<>(currentS3Listing);
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Can't connect to the bucket");
+            logger.debug("Can't connect to the bucket: {}", e.getMessage());
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> executionJob = this.executionJob;
+        if (executionJob != null) {
+            executionJob.cancel(true);
+        }
+    }
```

Comment:
Not sure, but i think you also need to set `this.executionJob = null;`

Reply:
Could you confirm this? Because if that's the case, the I need to add this to the other 2 things.

Reply:
Checked several other bindings and about all of them have the ScheduledFuture set to null in the  dispose method. So please adjust.

Reply:
Added for new and existing things